### ISSUE TITLE
How to limit years?

### ISSUE THREAD

**Issue by kartiksolanki** [twinssbc/Ionic2-Calendar#518]
I am using monthview for my calendar and I want to restrict the previous and next year user can swipe too. For example, I just want user to see all the events starting from Jan 2020 and until Dec 2023. Is it possible to restrict years?

**Comment by twinssbc** [comment]
@kartiksolanki 
There is lockSwipeToPrev option that you could set it when the current month goes to Jan 2020.
Unfortunately there's no lockSwipeToNext option, because last time when I want to add it, I found it's not working as expected in the underlying Swiper component. Let me check again to see if the issue is resolved now.

**Comment by kartiksolanki** [comment]
I did this if condition:
if (event.getFullYear() < this.calendar.currentDate.getFullYear()) { 
this.lockSwipeToPrev = true;
}

And the lock swipe works but it locks on December 2019 not Janurary 2020. I don't want user to go to December 2019. I just want to lock it at January 2020. Any idea what needs to be done?

**Comment by twinssbc** [comment]
@kartiksolanki Then you have to also check month, if year is 2020, month is January, disable the swipe.

**Comment by kartiksolanki** [comment]
@twinssbc Perfect. That worked like a charm. Thanks a lot. Please let me know if you can find a solution for "lockSwipeToNext".

**Comment by twinssbc** [comment]
@kartiksolanki I did a test with lockSwipeToNext function, it seems the behavior is still not expected. Actually I found even **lockSwipeToPrev** function has the same problem. As long as I set lockSwipeToXXX option to true, after swiping several slides, you will find that some slides don't update automatically. Only after you hovering your mouse/finger, the slide will refresh. This will cause confusion to the user. I suspect there's some bug in the underlying Swiper component. When allowSwipeXXX is turned on, the cached inactive slides don't auto refresh.

In summary, I won't add lockWipeToNext option, I also don't recommend use lockSwipeToPrev option.

**Comment by kartiksolanki** [comment]
@twinssbc Thanks.
